File: src/136.py
# y(4a-y) = n, y > a
def f(max_n):
    d = [0]*(max_n+1)
    for y in range(1, max_n+1):
        if y%100000 == 0: print(y)
        for a in range(y//4, y):  # Skip entire neg range
            x = y*(4*a-y)
            if x > max_n: break
            if x > 0:
                d[x] += 1

    return d

print(f(50000000).count(1))

File: src/136.cpp
#include <iostream>
#include <algorithm>

const int MAX_N = 50000000;
char d[MAX_N+1];

int main()
{
    for (int y=1; y<=MAX_N; y++)
    {
        for (int a=y/4; a<y && y*(4*a-y) <= MAX_N; a++)
        {
            if (4*a-y > 0) d[y*(4*a-y)]++;
        }
    }
    std::cout << std::count(d, d+MAX_N+1, 1) << std::endl;
    return 0;
}
